src/renderer/components/contact/ContactList.tsx

```diff
@@ -118,7 +118,9 @@ export function useContactsMap(listFlags: number, queryStr: string) {
   return [contacts, updateContacts] as [typeof contacts, typeof updateContacts]
 }
 
-export function useContactsNew(listFlags: number, initialQueryStr: string) {
+
+// if trim is provided and is true, the queryStr is trimmed for contact search and for checking validity of the string as an Email address
+export function useContactsNew(listFlags: number, initialQueryStr: string, trim?: boolean) {
```

Comment:
I think you can just trim it in all cases, this does not need to be optional, as @link2xt said we can also implement that in core, which would mean the input is trimmed on every call to `getContacts` and `checkEmailValidity`

Reply:
As link2xt said, I also feel doing it here rather than in the core is a better idea. I agree with `useContactsNew` always trimming as it is called only for CreateChat dialogue.